qemu/tests/nic_hotplug.py

```diff
@@ -140,12 +140,20 @@ def run_sub_test(params, plug_tag):
                  if needed
         """
         sub_type = params.get("sub_type_%s" % plug_tag)
+        session = vm.wait_for_login(timeout=params.get_numeric("login_timeout"))
+        shutdown_method = params.get("shutdown_method", "shell")
+        shutdown_command = params.get("shutdown_command")
         if sub_type:
```

Comment:
All right, remove `run_virt_sub_test` is a good change.
However, sub_type can be `boot` and `shutdown`, and they are independent. so your need to check the type of sub_type at first, and then do the next step.

```
if sub_type == 'boot':
    xxxx
else:
    xxxx
````

Reply:
> All right, remove `run_virt_sub_test` is a good change.
> However, sub_type can be `boot` and `shutdown`, and they are independent. so your need to check the type of sub_type at first, and then do the next step.
> 
> ```
> if sub_type == 'boot':
>     xxxx
> else:
>     xxxx
> ```
Hi @PaulYuuu Thanks for your help. I have updated it, please help me review it again, thanks a lot.